File: Solutions in JAVA/two-sum.java
import java.util.*;

class Program {
  public static int[] twoNumberSum(int[] array, int targetSum) {
		HashSet<Integer> uniqueNums = new HashSet<Integer>();
		for (int i= 0; i < array.length; i++){
			int requiredNum = targetSum - array[i];
			if (uniqueNums.contains(requiredNum)){
				return new int[] {array[i], requiredNum};
			}
			if (!uniqueNums.contains(array[i])){
					uniqueNums.add(array[i]);
			}
		}
		
		return new int[0];
		}
	}

File: Solutions In Python/two-sum.py
def twoNumberSum(array, targetSum):
    unique_numbers = set()
    for num in array:
        if num not in unique_numbers:
            unique_numbers.add(num)
            required_num = targetSum - num
        if required_num == num:
            continue
        elif required_num in unique_numbers:
            return [num, required_num]
    
    return []
		
